README.md

```diff
@@ -22,4 +22,31 @@ A Snap plugin is a program that responds to a set of well defined [gRPC](http://
 You will find [example plugins](examples) that cover the basics for writing collector, processor, and publisher plugins in the examples folder.
 
 
-
+## Snap Diagnostics
+Snap plugins using plugin-lib-go can be run independent of Snap to show their current running diagnostics. This diagnostic information includes:
+* Warning if dependencies not met
+* Config policy
+    * Warning if config items required and not provided
+* Collectable metrics
+* Runtime details
+    * Plugin version
+    * RPC type and version
+    * OS, architecture
+    * Golang version
+* How long it took to run each of these diagnostics
+
+### Running Diagnostics
+Running plugin diagnostics is easy! Simply build the plugin, then run the executable `$./build/${GOOS}/${GOARCH}/<plugin binary>`. When ran on its own, it will show a warning if a config is required for the plugin to load. 
+
+### Global Flags
+For specific details and to see all the options when running, run the plugin with the `-help` flag. The flag options are:
+```--config value       config file to use
+   --port value         port to listen on
+   --pingTimeout value  how much time must elapse before a lack of Ping results in a timeout (default: 0s)
```

Comment:
Global flags in doc seem to be outdated. 

Currently it looks like below:
```
GLOBAL OPTIONS:
   --config value            config file to use
   --port value              port to listen on
   --pprof                   set pprof
   --tls                     enable TLS
   --cert-path value         necessary to provide when TLS enabled
   --key-path value          necessary to provide when TLS enabled
   --root-cert-paths value   root paths separated by OS-specific list separator (e.g. 58)
   --stand-alone             enable stand alone plugin
   --stand-alone-port value  specify http port when stand-alone is set (default: 8181)
   --log-level value         log level - 0:panic 1:fatal 2:error 3:warn 4:info 5:debug (default: 2)
   --required-config         Plugin requires config passed in
   --help, -h                show help
   --version, -v             print the version
```

By the way,  a separator of root-cert-paths should be presented in differently:
--root-cert-paths value   root paths separated by OS-specific list separator (e.g. **58**)